variable_test.go

```diff
@@ -93,9 +114,22 @@ func TestVariablesCreate(t *testing.T) {
 		assert.NotEmpty(t, v.ID)
 		assert.Equal(t, *options.Key, v.Key)
 		assert.Equal(t, *options.Value, v.Value)
+		assert.Equal(t, *options.Description, v.Description)
 		assert.Equal(t, *options.Category, v.Category)
 	})
 
+	t.Run("when options has a too-long description", func(t *testing.T) {
+		options := VariableCreateOptions{
+			Key:         String(randomString(t)),
+			Value:       String(randomString(t)),
+			Description: String("tortor aliquam nulla facilisi cras fermentum odio eu feugiat pretium nibh ipsum consequat nisl vel pretium lectus quam id leo in vitae turpis massa sed elementum tempus egestas sed sed risus pretium quam vulputate dignissim suspendisse in est ante in nibh mauris cursus mattis molestie a iaculis at erat pellentesque adipiscing commodo elit at imperdiet dui accumsan sit amet nulla facilisi morbi tempus iaculis urna id volutpat lacus laoreet non curabitur gravida arcu ac tortor dignissim convallis aenean et tortor"),
+			Category:    Category(CategoryTerraform),
+		}
+
+		_, err := client.Variables.Create(ctx, wTest.ID, options)
+		assert.EqualError(t, err, "Invalid Attribute\n\nDescription is too long (maximum is 512 characters)")
```

Comment:
i think we might not want to check for the exact error message returned from the api here.

from what i can tell, the existing tests only check for a specific error message for error messages that go-tfe is returning. i think this makes sense because if the error message changes in the api, this test will break. 

i do think you are right about checking this case though. i think you could check that an error is returned from the api, and not check specifically what the error message is. there are a few other tests that do that. the workspace tests for "when an error is returned from the api" or the user test "with invalid email address" do this.